### PR TITLE
Javascript Methods - Michael Van Doorn

### PR DESCRIPTION
**Pull request by Michaelvxd**
CleanShot 2023-01-03 at 19 36 35@2x

Why do we log things to the console?
- Most often to check the current value being held in a variable

What is a method?
- A function that is a property of an object.

How is a method different from any other function?
- A method is a function that is associated with an object, while a function is a standalone piece of code that is not associated with any particular object.

How do you remove the last element from an array?
-  the .pop() method

How do you round a number down to the nearest integer?
- the Math.floor() method with your variable as the argument

How do you generate a random number?
- the Math.random() method with your variable as the argument. 

How do you delete an element from an array?
- The .shift() or .splice() methods will allow you to delete an element.

How do you append an element to an array?
- the .push() or .unshift() methods

How do you break a string up into an array?
- The .split() method will allow you to replace to insert an element.

Do string methods change the original string? How would you check if you weren't sure?
- No. For example .split creates a new array. You can check via console.log

Roughly how many string methods are there according to the MDN Web docs?
- 50 but many are deprecated 

Is the return value of a function or method useful in every situation?
- No. Sometimes you do not want to return a value (for example calling another function). 

Roughly how many array methods are there according to the MDN Web docs?
- ~40

What three-letter acronym should you always include in your Google search about a JavaScript method or CSS property?
- MDN

### CODE REVIEW

File: javascript-methods/main.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  author: 'Lea Verou'
+};
+
+library.push(js);
+library.unshift(css);
+library.splice(1, 1);
+
+console.log('library:', library);
+
+var fullName = 'Michael VanDoorn';
```
Comment (Michaelvxd) [inline javascript-methods/main.js:50]: 😆 Curious how I could reference 'Van Doorn' as a single index in an array as my last name using .split being that my true last name is two-part

Comment (Michaelvxd) [reply javascript-methods/main.js:50]: I guess .split and then concatenate the two array indexes but doesn't seem pretty?

File: javascript-methods/main.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+var numOne = 47;
+var numtwo = 23;
+var numThree = 103;
+
+var maximumValue = Math.max(numOne, numtwo, numThree);
+console.log(`maximumValue: ${maximumValue}`);
+
+var heroes = ['Batman', 'Superman', 'The Flash', 'Blackwidow'];
+var randomNumber = Math.random(heroes);
```
Comment (kurtyazdizadeh) [inline javascript-methods/main.js:9]: passing `heroes` into the `random` method has no effect. It doesn't have any parameters: [Math.random() doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random)

File: javascript-methods/main.js (hunk 1 above)
Comment (kurtyazdizadeh) [reply javascript-methods/main.js:50]: Interesting. There are definitely a few ways to approach this...

One options is to look character codes like this one `\xa0` to add a non-breaking space to your last name like this `Van\xa0Doorn`

look at the first answer in this stack overflow post: https://stackoverflow.com/questions/5237989/how-is-a-non-breaking-space-represented-in-a-javascript-string

**Review by kurtyazdizadeh — CHANGES_REQUESTED** [review]
Just remove the argument in `Math.random()`

File: javascript-methods/main.js (hunk 2 above)
Comment (Michaelvxd) [reply javascript-methods/main.js]: Fixed ✅ !

File: javascript-methods/main.js (hunk 1 above)
Comment (Michaelvxd) [reply javascript-methods/main.js]: Ahhh that'll work perfectly.  Thank you!

The curse of a two-part last name comes up everywhere hah

**Review by kurtyazdizadeh — APPROVED** [review]
Approved! 👍

**Pull request merged**